fdm-app/app/root.tsx

```
@@ -107,10 +122,17 @@ export function Layout() {
                 <ErrorBoundary error={null} params={{}} />
                 <ScrollRestoration
                     getKey={(location) => {
-                        // Use pathname for scroll restoration
                         return location.pathname
                     }}
                 />
+                {/* Inject runtime environment variables */}
+                {runtimeEnv && (
+                    <script
+                        dangerouslySetInnerHTML={{
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)